server/sonar-server/src/main/java/org/sonar/server/webhook/WebhookPayloadFactoryImpl.java

```diff
@@ -123,9 +123,12 @@ private String branchUrlOf(Project project, Branch branch) {
       }
       return format("%s/dashboard?branch=%s&id=%s",
         server.getPublicRootUrl(), encode(branch.getName().orElse("")), encode(project.getKey()));
-    } else {
+    } else if (branch.getType() == Branch.Type.SHORT) {
```

Comment:
suggestion: use case switch and throw an fail (silently?) if the branch type is not handled